src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -557,6 +556,26 @@ public static IResourceBuilder<ProjectResource> DisableForwardedHeaders(this IRe
         return builder;
     }
 
+    /// <summary>
+    /// Call a filter on each endpoint to determine if it should be included in the environment injection.
+    /// By default, all endpoints are included (if this method is not called).
+    /// </summary>
+    /// <param name="builder">The project resource builder.</param>
+    /// <param name="filter">The filter callback that returns true if and only if the endpoint should be included.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<ProjectResource> WithEndpointsInEnvironment(
+        this IResourceBuilder<ProjectResource> builder, Func<EndpointAnnotation, bool> filter)
+    {
+        foreach (var e in builder.Resource.Annotations.OfType<EndpointAnnotation>())
+        {
+            if (!filter(e))
+            {
+                builder.Resource.SkipEndpointEnvironment(e);
+            }
+        }
```

Comment:
I think we want to store the filter and have it run when the environment is being evaluated.

Reply:
Interesting. Is the reasoning that the state of the `EndpointAnnotation` could change between the call to `WithEndpointsInEnvironment` and the env evaluation, such that the user filter would execute differently before and after?

Reply:
Well, regardless of whether it makes a semantic difference, your suggesting makes the implementation cleaner, so I changed it accordingly.